### PR TITLE
Add support for cargo --profile

### LINKED ISSUE #167: Support custom cargo profiles
Now that https://github.com/rust-lang/cargo/issues/6988 is stabilized, it might be nice if it were possible to specify any profile rather than just --dev or --release. This was we can have an optimised debug release which is probably optimal for flamegraphing:

```
[profile.release-debug]
lto = "fat"
codegen-units = 1
debug = true
```

### PR DESCRIPTION
**Pull request by madadam**
Closes #167

### CODE REVIEW

**Comment by madadam** [comment]
Question: should we also deprecate / remove the `--dev` flag?

File: src/bin/cargo-flamegraph.rs [hunk 1]
```diff
@@ -84,9 +88,13 @@ fn build(opt: &Opt, kind: impl IntoIterator<Item = String>) -> anyhow::Result<Ve
         cmd.arg("build");
     }
 
-    // do not use `--release` when we are building for `bench`
-    if !opt.dev && opt.bench.is_none() {
-        cmd.arg("--release");
+    if let Some(profile) = &opt.profile {
+        cmd.arg("--profile").arg(profile);
+    } else {
```
Comment (djc) [inline src/bin/cargo-flamegraph.rs:93]: Would prefer to use `else if` directly here.

File: src/lib.rs [hunk 2]
```diff
@@ -501,7 +501,7 @@ pub struct FlamegraphOptions {
     pub notes: Option<String>,
 
     /// Omit functions smaller than <FLOAT> pixels
-    #[clap(long, default_value = "0.01f32", value_name = "FLOAT")]
+    #[clap(long, default_value = "0.01", value_name = "FLOAT")]
```
Comment (djc) [inline src/lib.rs:504]: This seems unrelated?

**Comment by djc** [comment]
> Question: should we also deprecate / remove the `--dev` flag?

No, let's keep that for now.

File: src/lib.rs (hunk 2 above)
Comment (madadam) [reply src/lib.rs:504]: Yes, it was throwing runtime error for me, probably because "0.01f32" doesn't parse to `f64`? But you are right, it's unrelated. Should I put it into a separate commit?

Comment (djc) [reply src/lib.rs:504]: Yes, splitting it out in a separate commit would be fine, thanks!

File: src/bin/cargo-flamegraph.rs [hunk 3]
```diff
@@ -84,8 +88,10 @@ fn build(opt: &Opt, kind: impl IntoIterator<Item = String>) -> anyhow::Result<Ve
         cmd.arg("build");
     }
 
+    if let Some(profile) = &opt.profile {
+        cmd.arg("--profile").arg(profile);
     // do not use `--release` when we are building for `bench`
```
Comment (djc) [inline src/bin/cargo-flamegraph.rs:93]: Not sure if this will pass `cargo fmt`, the comment placement seems a bit weird here.

Comment (madadam) [reply src/bin/cargo-flamegraph.rs:93]: This is actually why I originally didn't collapse the `else if`.

Comment (madadam) [reply src/bin/cargo-flamegraph.rs:93]: btw this is already has `cargo fmt` applied

Comment (djc) [reply src/bin/cargo-flamegraph.rs:93]: Okay, let's just move the comment inside the `else if`? That seems like a cleaner way to clarify.

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Thanks!
